python/CMakeLists.txt

```diff
@@ -1,27 +1,163 @@
+   runtime/common/py_ExecutionContext.cpp
+   runtime/cudaq/algorithms/py_sample_async.cpp
+   runtime/cudaq/algorithms/py_observe_async.cpp
+   runtime/cudaq/platform/JITExecutionCache.cpp
+   utils/LinkedLibraryHolder.cpp
+   runtime/cudaq/target/py_runtime_target.cpp
+   runtime/cudaq/target/py_testing_utils.cpp
+   runtime/common/py_SampleResult.cpp
+   runtime/common/py_ObserveResult.cpp
+   runtime/common/py_NoiseModel.cpp
+   runtime/cudaq/spin/py_matrix.cpp
+   runtime/cudaq/spin/py_spin_op.cpp 
+   runtime/cudaq/algorithms/py_optimizer.cpp 
+   ../runtime/cudaq/platform/common/QuantumExecutionQueue.cpp
+
+  EMBED_CAPI_LINK_LIBS
+   CUDAQuantumMLIRCAPI
+   MLIRCAPIExecutionEngine
+  PRIVATE_LINK_LIBS
+   OptCodeGen
+   OptTransforms
+)
+
+target_include_directories(CUDAQuantumPythonSources.Extension INTERFACE 
+    ${CMAKE_CURRENT_SOURCE_DIR} 
+    ${CMAKE_SOURCE_DIR}/python 
+    ${CMAKE_SOURCE_DIR}/python/utils)
+target_link_libraries(CUDAQuantumPythonSources.Extension INTERFACE 
+    cudaq 
+    cudaq-common 
+    cudaq-em-default
+    fmt::fmt-header-only
+    )
+
+################################################################################
+# Common CAPI
+################################################################################
+
+add_mlir_python_common_capi_library(CUDAQuantumPythonCAPI
+  INSTALL_COMPONENT CUDAQuantumPythonModules
+  INSTALL_DESTINATION mlir_cudaq/_mlir_libs
+  OUTPUT_DIRECTORY "${MLIR_BINARY_DIR}/python/mlir_cudaq/_mlir_libs"
+  RELATIVE_INSTALL_ROOT "../.."
+  DECLARED_SOURCES
+    CUDAQuantumPythonSources
+    # TODO: Remove this in favor of showing fine grained registration once
+    # available.
+    MLIRPythonExtension.RegisterEverything
+    MLIRPythonSources.Core
+)
+
+################################################################################
+# Instantiation of all Python modules
+################################################################################
+
+add_mlir_python_modules(CUDAQuantumPythonModules
+  ROOT_PREFIX "${MLIR_BINARY_DIR}/python/mlir_cudaq"
+  INSTALL_PREFIX "mlir_cudaq"
+  DECLARED_SOURCES
+    CUDAQuantumPythonSources
+    # TODO: Remove this in favor of showing fine grained registration once
+    # available.
+    MLIRPythonExtension.RegisterEverything
+    MLIRPythonSources
+  COMMON_CAPI_LINK_LIBS
+    CUDAQuantumPythonCAPI
+  )
+
+file (COPY cudaq DESTINATION ${MLIR_BINARY_DIR}/python)
+
+## The Python bindings module for Quake dialect depends on CUDAQ libraires 
+## which it can't locate since they are in "../../lib" and the 'rpath' is set
+## to '$ORIGIN' by default.
+set(_origin_prefix "\$ORIGIN")
+if(APPLE)
+  set(_origin_prefix "@loader_path")
+endif()
+set_property(TARGET CUDAQuantumPythonModules.extension._quakeDialects.dso 
+             APPEND PROPERTY INSTALL_RPATH "${_origin_prefix}:${_origin_prefix}/../../lib/")
+
+
+## TODO: Clean-up following from 'old' CMakeLists.txt
+## FIXME: Delete '_pycudaq' target?
```

Comment:
Yea I think this can all go away. Also there are some old c++ files that aren't going to be used any more, like py_sample.cpp

Reply:
Added to https://github.com/NVIDIA/cuda-quantum/issues/1157